cmd/tailscale/cli/file.go

```diff
@@ -286,22 +287,109 @@ func runCpTargets(ctx context.Context, args []string) error {
 	return nil
 }
 
+type onConflict string
+
+const (
+	skipOnExist         onConflict = "skip"
+	overwriteExisting   onConflict = "overwrite" //  Overwrite any existing file at the target location
+	createNumberedFiles onConflict = "rename"    //  Create an alternately named file in the style of Chrome Downloads
+)
+
+func (v *onConflict) String() string {
+	return string(*v)
+}
+
+func (v *onConflict) Set(s string) error {
+	if s == "" {
+		*v = skipOnExist
+		return nil
+	}
+	*v = onConflict(strings.ToLower(s))
+	if *v != skipOnExist && *v != overwriteExisting && *v != createNumberedFiles {
+		return fmt.Errorf("%q is not one of (skip|overwrite|rename)", s)
+	}
+	return nil
+}
+
 var fileGetCmd = &ffcli.Command{
 	Name:       "get",
-	ShortUsage: "file get [--wait] [--verbose] <target-directory>",
+	ShortUsage: "file get [--wait] [--verbose] [--conflict=(skip|overwrite|rename)] <target-directory>",
 	ShortHelp:  "Move files out of the Tailscale file inbox",
 	Exec:       runFileGet,
 	FlagSet: (func() *flag.FlagSet {
 		fs := newFlagSet("get")
 		fs.BoolVar(&getArgs.wait, "wait", false, "wait for a file to arrive if inbox is empty")
 		fs.BoolVar(&getArgs.verbose, "verbose", false, "verbose output")
+		fs.Var(&getArgs.conflict, "conflict", "behavior when target file already exists")
```

Comment:
I ran this and the output is:

```
USAGE
  file get [--wait] [--verbose] [--conflict=(skip|overwrite|rename)] <target-directory>

FLAGS
  -conflict skip  behavior when target file already exists
  -verbose=false  verbose output
  -wait=false     wait for a file to arrive if inbox is empty
```

I think the conflict flag's docs should be fleshed out here. Maybe:

`"how to handle conflicting filenames: \"overwrite\" to overwrite the local existing file, \"rename\" to make a new number-suffixed local name that's unique, or \"skip\" to ....`

Actually, what does "skip" even do? It looks like it fails and doesn't skip?

Ah, it gathers accumulated errors and prints them all, but doesn't delete the waiting file from the "inbox" area.

Reply:
Maybe "error" instead of "skip"? 🤔